huntsman/observatory.py

```diff
@@ -512,3 +363,159 @@ def _create_flat_field_observation(self, alt=None, az=None,
+        """Take flat fields.
+
+        Args:
+            alt (float, optional): Altitude for flats
+            az (float, optional): Azimuth for flats
+            min_counts (int, optional): Minimum ADU count
+            max_counts (int, optional): Maximum ADU count
+            bias (int, optional): Default bias for the cameras
+            max_exptime (float, optional): Maximum exposure time before stopping
+            camera_list (list, optional): List of cameras to use for flat-fielding
+            target_adu_percentage (float, optional): Exposure time will be adjust so
+                that counts are close to: target * (`min_counts` + `max_counts`). Default
+                to 0.5
+            max_num_exposures (int, optional): Maximum number of flats to take
+            *args (TYPE): Description
+            **kwargs (TYPE): Description
+
+        """
+        target_adu = target_adu_percentage * (min_counts + max_counts)
+        exptimes = {cam_name: [1. * u.second] for cam_name in camera_names}
+
+        # Create the observation objects
+        observations = {}
+        for cam_name in camera_names:
+            observations[cam_name] = self._create_flat_field_observation(
+                        alt=alt, az=az, filter_name=filter_names[cam_name])
+
+        # Loop until conditions are met to finish flat-fielding
+        exposure_count = 0
+        while True:
+
+            # Update the exposure count
+            exposure_count += 1
+
+            # Get the FITS headers
+            fits_headers = {cam_name: self.get_standard_headers(
+                observation=observations[cam_name]) for cam_name in camera_names}
+
+            # Use a common start time for all cameras
+            start_time = utils.current_time()
+            for cam_name in camera_names:
+                fits_headers[cam_name]['start_time'] = utils.flatten_time(start_time)
+
+            # Take the flat field observations (blocking)
+            _exptimes = {cam_name: exptimes[cam_name][-1] for cam_name in camera_names}
+            camera_events = self._take_flat_fields(
+                        camera_names, observations, _exptimes, fits_headers)
+
+            # Check the counts for each image
+            is_saturated = False
+            for cam_name, info in camera_events.items():
+                filename = info['filename']
+                self.logger.debug("Checking counts for {}".format(filename))
+
+                # Unpack fits if compressed
+                if not os.path.exists(filename) and \
+                        os.path.exists(filename.replace('.fits', '.fits.fz')):
+                    fits_utils.fpack(filename.replace('.fits', '.fits.fz'), unpack=True)
+
+                # Calculate average counts per pixel
+                data = fits.getdata(filename)
+                mean_counts, _, _ = stats.sigma_clipped_stats(data)
+                mean_counts -= bias
```

Comment:
This construct sometimes doesn't work if `mean_counts` and `bias` have different data types. 

```suggestion
                mean_counts = mean_counts - bias
```